src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```
@@ -12,11 +15,32 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+                throw new DataProcessingException("Can't save actor to DB" + actor, e);
+            }
```

There is an issue with exception handling here. According to the checklist, you should throw the `DataProcessingException` regardless of whether the transaction is null. Currently, if the transaction is null, the exception will not be thrown, and the error will be silently ignored. Move the `throw new DataProcessingException(...)` outside of the `if (transaction != null)` block.